src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/PEReaderExtensions.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+
+namespace ILCompiler.Reflection.ReadyToRun
+{
+    public class PEExportTable
+    {
+        private readonly Dictionary<string, int> _namedExportRva;
+        private readonly Dictionary<int, int> _ordinalRva;
+
+        private PEExportTable(PEReader peReader)
+        {
+            _namedExportRva = new Dictionary<string, int>();
+            _ordinalRva = new Dictionary<int, int>();
+
+            DirectoryEntry exportTable = peReader.PEHeaders.PEHeader.ExportTableDirectory;
+            PEMemoryBlock peImage = peReader.GetEntireImage();
+            BlobReader exportTableHeader = peImage.GetReader(peReader.GetOffset(exportTable.RelativeVirtualAddress), exportTable.Size);
+            if (exportTableHeader.Length == 0)
+            {
+                return;
+            }
+
+            // +0x00: reserved
+            exportTableHeader.ReadUInt32();
+            // +0x04: TODO: time/date stamp
+            exportTableHeader.ReadUInt32();
+            // +0x08: major version
+            exportTableHeader.ReadUInt16();
+            // +0x0A: minor version
+            exportTableHeader.ReadUInt16();
+            // +0x0C: DLL name RVA
+            exportTableHeader.ReadUInt32();
+            // +0x10: ordinal base
+            int minOrdinal = exportTableHeader.ReadInt32();
+            // +0x14: number of entries in the address table
+            int addressEntryCount = exportTableHeader.ReadInt32();
+            // +0x18: number of name pointers
+            int namePointerCount = exportTableHeader.ReadInt32();
+            // +0x1C: export address table RVA
+            int addressTableRVA = exportTableHeader.ReadInt32();
+            // +0x20: name pointer RVA
+            int namePointerRVA = exportTableHeader.ReadInt32();
+            // +0x24: ordinal table RVA
+            int ordinalTableRVA = exportTableHeader.ReadInt32();
+
+            int[] addressTable = new int[addressEntryCount];
```

Comment:
Numbers in PE files are in little-endian format, but BlobReader.ReadInt32() doesn't seem to take endianess into consideration.  Will this code work in big-endian machines too?

Reply:
Certainly not. I have no idea what is our big endian story, last time I heard there was none.

Reply:
> BlobReader.ReadInt32() doesn't seem to take endianess into consideration.

It does: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobReader.cs#L275

CoreCLR does not target any big endian platform currently, and so there is number of bugs that would need to be fixed if that changes.